### ISSUE TITLE
[ENH]: Update grid_helper on FloatingSubplot

### ISSUE THREAD

**Issue by albavilanova** [matplotlib/matplotlib#25426]
### Problem

Good afternoon,

I created a plot using FloatingSubplot and I had to define the grid_helper without having the extremes and grid locators.  These are calculated later in my code, and I would like to be able to update the grid_helper (or the extremes and grid locators) on my axis once I have the correct values. I couldn't find a function to do so, could this be a new feature?

Thank you in advance,
Alba

**Comment by anntzer** [comment]
Once #25427 gets merged, I think the following works (starting from the demo_floating_axes.py example):
```python
ax1.get_grid_helper().update_grid_finder(
    extreme_finder=floating_axes.ExtremeFinderFixed((0, 3, -1, 5)))
ax1.clear()
ax1.adjust_axes_lim()
```
Unfortunately it is for now necessary to call clear() (and adust_axes_lim, which could reasonably be folded into clear()); it's *probably*(?) possible to get rid of that and just have a self-contained method on the axes that updates just whatever is needed but that'll require a bit of careful surgery.

**Comment by albavilanova** [comment]
Thanks @anntzer! I will test it once it is merged. I should be able to add `grid_locator1` as an argument in `update_grid_finder` too, right?

**Comment by anntzer** [comment]
Perhaps?  Please give it a try and let us know what works and what doesn't.

**Comment by oscargus** [comment]
#25427 is now merged in main.

**Comment by Higgs32584** [comment]
Is this issue resolved?

**Comment by albavilanova** [comment]
I haven't tested it on our software yet since I am not allowed to install any dependencies on our machines, and I am waiting for my request at work to be approved. Feel free to close the issue, if something is wrong I will inform you and reopen it. Thank you all!

**Comment by Higgs32584** [comment]
Oh I can't close stuff I haven't gotten my first commit yet 😂😂

**Comment by albavilanova** [comment]
Just to let you know, the extremes are updating perfectly now. Many thanks!
